src/rest_api/classes/protein/widgets/sequences.clj

```diff
@@ -1,21 +1,61 @@
 (ns rest-api.classes.protein.widgets.sequences
   (:require
-    [rest-api.classes.generic-fields :as generic]))
+    [clojure.java.jdbc :as jdbc]
+    [clojure.string :as str]
+    [rest-api.db.sequence :as wb-seq]
+    [rest-api.classes.generic-fields :as generic])
+  (:import (org.biojava.nbio.core.sequence ProteinSequence DNASequence)
+           (org.biojava.nbio.aaproperties PeptideProperties)))
+
+(def abbrev
+  {"A" "Ala"
+   "R" "Arg"
+   "N" "Asn"
+   "D" "Asp"
+   "C" "Cys"
+   "E" "Glu"
+   "Q" "Gln"
+   "G" "Gly"
+   "H" "His"
+   "I" "Ile"
+   "L" "Leu"
+   "K" "Lys"
+   "M" "Met"
+   "F" "Phe"
+   "O" "Pyl*"
+   "P" "Pro"
+   "S" "Ser"
+   "T" "Thr"
+   "W" "Trp"
+   "Y" "Tyr"
+   "V" "Val"
+   "U" "Sec*"
+   "X" "Xaa**"})
 
 (defn estimated-isoelectric-point [p]
-  {:data nil ; should use org.biojava.bio.proteomics.IsoelectricPointCalc see: http://biojava.org/wiki/BioJava%3ACookbook%3AProteomics
+  {:data (when-let [h (:protein/peptide p)]
+           (. PeptideProperties getIsoelectricPoint (:peptide/sequence (:protein.peptide/peptide h))))
    :description "the estimated isoelectric point of the protein"})
 
 (defn estimated-molecular-weight [p]
   {:data (when-let [mw (:protein.molecular-weight/float
-           (first
-             (:protein/molecular-weight p)))]
+                         (first
+                           (:protein/molecular-weight p)))]
            (format "%.1f" mw))
    :description "the estimated molecular weight of the protein"})
 
 (defn amino-acid-composition [p]
-  {:data (keys p)
-   :d (:db/id p) ; bioperl uses seqStats
+  {:data 
+   (let [source (when-let [h (:protein/peptide p)]
+                  (->(:peptide/sequence (:protein.peptide/peptide h))
+                                        (str/split  #"")
```

Comment:
Some indentation is a bit off here and in a few other place. Can you configure your text editor to reformat the code using indentation rules for Clojure?

Here is the style guide: https://github.com/bbatsov/clojure-style-guide

Reply:
Ah, that's weird.  Adam helped me set up vim to auto-indent by highlighting everything and pressing = .  I think the lack of a space between (-> and (:peptide/sequence was messing that up.

I've committed another change for "Indentation changes" to the repo